packages/host/app/services/loader-service.ts

```diff
@@ -32,8 +41,70 @@ export default class LoaderService extends Service {
       new URL(baseRealm.url),
       new URL(config.resolvedBaseRealmURL),
     );
+    loader.prependURLHandlers([(req) => this.fetchWithAuth(req)]);
     shimExternals(loader);
 
     return loader;
   }
+
+  private async fetchWithAuth(request: Request) {
+    // To avoid deadlock, we can assume that any GET requests to baseRealm don't need authentication.
+    let isGetRequestToBaseRealm =
+      request.url.includes(baseRealm.url) && request.method === 'GET';
+    if (
+      isGetRequestToBaseRealm ||
+      request.url.includes('session') ||
```

Comment:
`request.url.includes('session')` will include any url with the word session in it - is that ok or did you mean the specific form of the realm's session endpoint?

Reply:
My idea is we don't need to include `Authorization` header, if the request is an authentication request. The way to differentiate it is by checking the `url`.

Reply:
And I have to update the realm checkPermission. [bcba1c2](https://github.com/cardstack/boxel/pull/1020/commits/bcba1c2f2a3979c1c36948bf2fa56b4678da1365)